sophus/rxso2.hpp

```diff
@@ -171,10 +171,7 @@ class RxSO2Base {
   ///
   SOPHUS_FUNC Tangent log() const {
     using std::log;
-    Tangent theta_sigma;
-    theta_sigma[1] = log(scale());
-    theta_sigma[0] = SO2<Scalar>(complex()).log();
-    return theta_sigma;
+    return (Tangent{} << SO2<Scalar>(complex()).log(), log(scale())).finished();
```

Comment:
I'm not too big of a fan of this syntax (or possibly just not familiar with it). Can we break it into two lines?